FreeBSD CI is failing (`Expected: ef99c96f929bc701f30cb113e1f5a1dee43f4f4d8ec0ec318a7ec2d02e5059e8 Received: d5442a6f1a84db8e787bbcfef53418146123ca4671ad674144d027fa87840577`)

https://github.com/containerd/nerdctl/actions/runs/6649372640/job/18068316744

```
2023-10-26T04:17:19.6348765Z ##[group]Run ln -sf Vagrantfile.freebsd Vagrantfile
2023-10-26T04:17:19.6349382Z [36;1mln -sf Vagrantfile.freebsd Vagrantfile[0m
2023-10-26T04:17:19.6349858Z [36;1msudo vagrant up --no-tty[0m
2023-10-26T04:17:19.6423303Z shell: /usr/bin/bash -e {0}
2023-10-26T04:17:19.6423670Z env:
2023-10-26T04:17:19.6424105Z   GO_VERSION: 1.21.x
2023-10-26T04:17:19.6424474Z ##[endgroup]
2023-10-26T04:17:31.4542007Z Bringing machine 'default' up with 'libvirt' provider...
2023-10-26T04:17:32.5759384Z ==> default: Box 'generic/freebsd13' could not be found. Attempting to find and install...
2023-10-26T04:17:32.5760729Z     default: Box Provider: libvirt
2023-10-26T04:17:32.5761477Z     default: Box Version: >= 0
2023-10-26T04:17:32.9667200Z ==> default: Loading metadata for box 'generic/freebsd13'
2023-10-26T04:17:32.9668653Z     default: URL: https://vagrantcloud.com/generic/freebsd13
2023-10-26T04:17:33.2232490Z ==> default: Adding box 'generic/freebsd13' (v4.3.4) for provider: libvirt
2023-10-26T04:17:33.2400736Z     default: Downloading: https://vagrantcloud.com/generic/boxes/freebsd13/versions/4.3.4/providers/libvirt/amd64/vagrant.box
2023-10-26T04:17:45.8425707Z 
2023-10-26T04:17:45.8428014Z     default: Calculating and comparing box checksum...
2023-10-26T04:17:48.4771314Z The checksum of the downloaded box did not match the expected
2023-10-26T04:17:48.4773138Z value. Please verify that you have the proper URL setup and that
2023-10-26T04:17:48.4775549Z you're downloading the proper file.
2023-10-26T04:17:48.4776064Z 
2023-10-26T04:17:48.4777729Z Expected: ef99c96f929bc701f30cb113e1f5a1dee43f4f4d8ec0ec318a7ec2d02e5059e8
2023-10-26T04:17:48.4779055Z Received: d5442a6f1a84db8e787bbcfef53418146123ca4671ad674144d027fa87840577
2023-10-26T04:17:48.4943415Z ##[error]Process completed with exit code 1.
```

--- a/Vagrantfile.freebsd
+++ b/Vagrantfile.freebsd
@@ -18,6 +18,8 @@
 # Vagrantfile for FreeBSD
 Vagrant.configure("2") do |config|
   config.vm.box = "generic/freebsd13"
+  # workaround for https://github.com/containerd/nerdctl/issues/2596
+  config.vm.box_version = "4.3.2"
 
   memory = 2048
   cpus = 1

--- a/cmd/nerdctl/compose_config_test.go
+++ b/cmd/nerdctl/compose_config_test.go
@@ -68,7 +68,11 @@ services:
 	comp := testutil.NewComposeDir(t, fmt.Sprintf(dockerComposeYAML, "3.13"))
 	defer comp.CleanUp()
 
-	base.ComposeCmd("-f", comp.YAMLFullPath(), "config", "--hash=*").AssertOutContains("hello1")
+	// `--hash=*` is broken in Docker Compose v2.23.0: https://github.com/docker/compose/issues/11145
+	if base.Target == testutil.Nerdctl {
+		base.ComposeCmd("-f", comp.YAMLFullPath(), "config", "--hash=*").AssertOutContains("hello1")
+	}
+
 	hash := base.ComposeCmd("-f", comp.YAMLFullPath(), "config", "--hash=hello1").Out()
 
 	newComp := testutil.NewComposeDir(t, fmt.Sprintf(dockerComposeYAML, "3.14"))

